pkg/middleware/create_home.go

```diff
@@ -0,0 +1,40 @@
+package middleware
+
+import (
+	"net/http"
+
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/rgrpc/status"
+	tokenpkg "github.com/cs3org/reva/pkg/token"
+	"google.golang.org/grpc/metadata"
+)
+
+// CreateHome provides a middleware which sends a CreateHome request to the reva gateway
+func CreateHome(opts ...Option) func(next http.Handler) http.Handler {
+	opt := newOptions(opts...)
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			token := r.Header.Get("x-access-token")
```

Comment:
this can potentially fail if OIDC is not configured on the proxy config, perhaps some defensive code might help future debugging